# Source: written by Anders Ladegaard Marchsteiner <alm.anma@gmail.com>
# Upstream: no, but solution found from this mailing list: http://mail-index.netbsd.org/current-users/2011/07/03/msg017003.html
# Reason: To fix build with newer curl.
diff -Naur a/lib/curl_transport/xmlrpc_curl_transport.c b/lib/curl_transport/xmlrpc_curl_transport.c
--- a/lib/curl_transport/xmlrpc_curl_transport.c	2011-12-03 21:56:02.622993493 +0000
+++ b/lib/curl_transport/xmlrpc_curl_transport.c	2011-12-03 21:59:26.193979573 +0000
@@ -83,7 +83,7 @@
 #include "version.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
+#include <curl/typecheck-gcc.h>
 #include <curl/easy.h>
 #include <curl/multi.h>
 
diff -rupN a/lib/curl_transport/curlmulti.c b/lib/curl_transport/curlmulti.c
--- a/lib/curl_transport/curlmulti.c	2012-01-15 16:02:51.677024799 +0000
+++ b/lib/curl_transport/curlmulti.c	2012-01-15 16:09:39.548004218 +0000
@@ -17,7 +17,7 @@
 #endif
 
 #include <curl/curl.h>
-#include <curl/types.h>
+#include <curl/typecheck-gcc.h>
 #include <curl/easy.h>
 #include <curl/multi.h>
 
diff -rupN a/lib/curl_transport/curltransaction.c b/lib/curl_transport/curltransaction.c
--- a/lib/curl_transport/curltransaction.c	2012-01-15 16:02:51.676024798 +0000
+++ b/lib/curl_transport/curltransaction.c	2012-01-15 16:09:22.696005127 +0000
@@ -15,7 +15,7 @@
 #include "version.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
+#include <curl/typecheck-gcc.h>
 #include <curl/easy.h>
 
 #include "curlversion.h"
diff -rupN a/lib/curl_transport/xmlrpc_curl_transport.c.orig b/lib/curl_transport/xmlrpc_curl_transport.c.orig
--- a/lib/curl_transport/xmlrpc_curl_transport.c.orig	2012-01-15 16:02:51.677024799 +0000
+++ b/lib/curl_transport/xmlrpc_curl_transport.c.orig	2012-01-15 16:09:00.707006202 +0000
@@ -83,7 +83,7 @@
 #include "xmlrpc-c/time_int.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
+#include <curl/typecheck-gcc.h>
 #include <curl/easy.h>
 #include <curl/multi.h>
 
